stl/inc/xstring

```diff
@@ -17,6 +17,10 @@
 #include <xpolymorphic_allocator.h>
 #endif // _HAS_CXX17
 
+#if _HAS_CXX20
+#include <compare>
+#endif // _HAS_CXX20
+

```

Comment:
`<xmemory>` includes `<xutility>` which includes `<utility>` which includes `<compare>` when `_HAS_CXX20`, so we don't need this direct include here.